File: 513-find-bottom-left-tree-value/513-find-bottom-left-tree-value.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun findBottomLeftValue(root: TreeNode?): Int {
        root ?: return -1
        var level = listOf(root)
        var leftMost = root.`val`
        while (level.isNotEmpty()) {
            leftMost = level.first().`val`
            level = level.flatMap { r -> listOf(r.left, r.right).filterNotNull() }
        }
        return leftMost
    }
}

File: 513-find-bottom-left-tree-value/513-find-bottom-left-tree-value.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        level, leftmost = [root], root.val
        while level:
            leftmost = level[0].val
            level = [c for n in level for c in [n.left, n.right] if c]
        return leftmost
        